package.json

```diff
@@ -107,7 +107,7 @@
     "@keystonehq/metamask-airgapped-keyring": "0.2.1",
     "@material-ui/core": "^4.11.0",
     "@metamask/contract-metadata": "^1.31.0",
-    "@metamask/controllers": "^22.0.0",
+    "@metamask/controllers": "^23.0.0",
```

Comment:
This update includes a breaking change to the approval controller, which is that an error is now explicitly required when calling `clear`. We call clear once in `metamask-controller.js` on line 537.

Reply:
Do you know what `code` and `message` we should use for the `EthereumRpcError` we pass?

Reply:
@rekmarks ☝️ do you know?

Reply:
That particular `clear` is triggered by the UI closing, which we interpret as the user rejecting the approval. So you can use `ethErrors.provider.userRejectedRequest()`, with no provided message. The default message is perfectly fine.

Reply:
added here: https://github.com/MetaMask/metamask-extension/pull/13110/commits/f18e1eea775cc11d5395071ec8294f2206b93fac